txpool/txpool.go

```diff
@@ -346,63 +338,46 @@ func (p *TxPool) Pop(tx *types.Transaction) {
 	}
 }
 
-// Drop removes the given (unrecoverable) transaction from
-// its associated promoted queue (account) and rolls
-// back the account's nextNonce.
-// Will update executables with the next primary
-// from that account (if any).
+// Drop clears the entire account associated with the given transaction
+// and reverts its next (expected) nonce.
 func (p *TxPool) Drop(tx *types.Transaction) {
+	// fetch associated account
 	account := p.accounts.get(tx.From)
 
 	account.promoted.lock(true)
-	defer account.promoted.unlock()
+	account.enqueued.lock(true)
 
-	// pop the top most promoted tx
-	account.promoted.pop()
+	defer func() {
+		account.enqueued.unlock()
+		account.promoted.unlock()
+	}()
 
-	// update state
-	p.index.remove(tx)
-	p.gauge.decrease(slotsRequired(tx))
+	// rollback nonce
+	nextNonce := tx.Nonce
+	account.setNonce(nextNonce)
 
-	// update metrics
-	p.metrics.PendingTxs.Add(-1)
+	// drop promoted
+	dropped := account.promoted.clear()
+	p.index.remove(dropped...)
+	p.gauge.decrease(slotsRequired(dropped...))
 
-	if tx.Nonce < account.getNonce() {
-		// rollback nonce
-		account.setNonce(tx.Nonce)
-	}
+	// update metrics
+	p.metrics.PendingTxs.Add(float64(-1 * len(dropped)))
 
-	// update executables
-	if tx := account.promoted.peek(); tx != nil {
-		p.executables.push(tx)
-	}
+	// drop enqueued
+	dropped = account.enqueued.clear()
+	p.index.remove(dropped...)
+	p.gauge.decrease(slotsRequired(dropped...))
 
 	p.eventManager.signalEvent(proto.EventType_DROPPED, tx.Hash)
+	p.logger.Debug("dropped account",
```

Comment:
Why is the message `dropped account`?
Shouldn't it be something like `dropped transaction`?

Reply:
Because we're dropping all transactions from that account. Perhaps a more detailed output on the actions of `Drop()` ?

Reply:
@lazartravica not just the enqueued transactions, but all transactions the account has (promoted + enqueued)

Reply:
@dbrajovic Got it, I'm still not at 100% with the new Txpool data structure glossary :)

 @zivkovicmilos please take a look at this thread

Reply:
Got it, maybe the message should be `dropped account transactions`

Reply:
[Refactored](https://github.com/0xPolygon/polygon-edge/pull/380/commits/8211ae5617010c1be5764e1069f43b37165ff177)

Reply:
How is it unambiguous now? 😄

Reply:
[Unambigous](https://github.com/0xPolygon/polygon-edge/pull/380#:~:text=fix%20log-,e48473c,-edit%20clearAccountQueue)